scripts/stac/imagery/tests/capture_area_test.py

```diff
@@ -0,0 +1,144 @@
+from shapely.geometry import Polygon
+
+from scripts.stac.imagery.capture_aera import generate_capture_area, merge_polygons, to_feature
+
+# In the following tests, the expected and result GeoJSON documents are printed if the test fails.
+# This allows to visualize the geometry for debugging purpose.
+
+
+def test_merge_polygons() -> None:
+    polygons = []
+    polygons.append(Polygon([(0.0, 1.0), (1.0, 1.0), (1.0, 0.0), (0.0, 0.0), (0.0, 1.0)]))
+    polygons.append(Polygon([(1.0, 1.0), (2.0, 1.0), (2.0, 0.0), (1.0, 0.0), (1.0, 1.0)]))
+    expected_merged_polygon = Polygon([(0.0, 1.0), (2.0, 1.0), (2.0, 0.0), (0.0, 0.0), (0.0, 1.0)])
+    print(f"GeoJSON expected: {to_feature(expected_merged_polygon)}")
+
+    merged_polygons = merge_polygons(polygons, 0)
+    print(f"GeoJSON result: {to_feature(merged_polygons)}")
+
+    # Using `Polygon.equals()` as merge_polygons might return a different set of coordinates for the same geometry
+    # In this example: `Polygon([(2.0, 1.0), (2.0, 0.0), (0.0, 0.0), (0.0, 1.0), (2.0, 1.0)])`
+    assert merged_polygons.equals(expected_merged_polygon)
+
+
+def test_merge_polygons_with_rounding() -> None:
+    polygons = []
+    polygons.append(Polygon([(0.0, 1.0), (1.0, 1.0), (1.0, 0.0), (0.0, 0.0), (0.0, 1.0)]))
+    # The following polygon is off by 0.1 to the "right" from the previous one
+    polygons.append(Polygon([(1.1, 1.0), (2.0, 1.0), (2.0, 0.0), (1.0, 0.0), (1.1, 1.0)]))
+    expected_merged_polygon = Polygon([(0.0, 1.0), (2.0, 1.0), (2.0, 0.0), (0.0, 0.0), (0.0, 1.0)])
+    print(f"GeoJSON expected: {to_feature(expected_merged_polygon)}")
+
+    # By giving a buffer distance of 0.1, we want to correct this margin of error and have the two polygons being merged
+    merged_polygons = merge_polygons(polygons, 0.1)
+    print(f"GeoJSON result: {to_feature(merged_polygons)}")
+
+    assert merged_polygons.equals(expected_merged_polygon)
+
+
+def test_merge_polygons_with_rounding_margin_too_big() -> None:
+    polygons = []
+    polygons.append(Polygon([(0.0, 1.0), (1.0, 1.0), (1.0, 0.0), (0.0, 0.0), (0.0, 1.0)]))
+    # The following polygon is off by 0.1 to the "right" from the previous one
+    polygons.append(Polygon([(1.1, 1.0), (2.0, 1.0), (2.0, 0.0), (1.0, 0.0), (1.1, 1.0)]))
+    expected_merged_polygon = Polygon([(0.0, 1.0), (2.0, 1.0), (2.0, 0.0), (0.0, 0.0), (0.0, 1.0)])
+    print(f"GeoJSON expected: {to_feature(expected_merged_polygon)}")
+
+    # Here, the allowed margin of error is 0.01 < 0.1
+    merged_polygons = merge_polygons(polygons, 0.01)
+    print(f"GeoJSON result: {to_feature(merged_polygons)}")
+
+    assert not merged_polygons.equals(expected_merged_polygon)
+
+
+def test_generate_capture_area_rounded() -> None:
+    polygons = []
+    polygons.append(
+        Polygon(
+            [
+                (178.254184985672936, -38.40856551170436),
```

Comment:
Would it be possible to simplify these polygons? I can't tell whether the test is correct because of the complexity of the inputs.